modules/programs/k9s.nix

```diff
@@ -140,31 +163,67 @@ in {
     };
   };
 
-  config = mkIf cfg.enable {
-    home.packages = [ cfg.package ];
-
-    xdg.configFile."k9s/config.yml" = mkIf (cfg.settings != { }) {
-      source = yamlFormat.generate "k9s-config" cfg.settings;
-    };
-
-    xdg.configFile."k9s/skin.yml" = mkIf (cfg.skin != { }) {
-      source = yamlFormat.generate "k9s-skin" cfg.skin;
-    };
-
-    xdg.configFile."k9s/aliases.yml" = mkIf (cfg.aliases != { }) {
-      source = yamlFormat.generate "k9s-aliases" cfg.aliases;
-    };
-
-    xdg.configFile."k9s/hotkey.yml" = mkIf (cfg.hotkey != { }) {
-      source = yamlFormat.generate "k9s-hotkey" cfg.hotkey;
-    };
-
-    xdg.configFile."k9s/plugin.yml" = mkIf (cfg.plugin != { }) {
-      source = yamlFormat.generate "k9s-plugin" cfg.plugin;
+  config = let
+    isAtLeast0_29 = (lib.versionAtLeast cfg.package.version "0.29.0");
+    isAtLeast0_30 = (lib.versionAtLeast cfg.package.version "0.30.0");
```

Comment:
I don't think you have to worry about these. The package in Nixpkgs unstable is (currently) 0.31.5 and I don't see any earlier version available there so in HM we shouldn't have to worry about earlier versions.

Reply:
Perfect, that simplifies things. I did wonder just how backwards-compatible we needed to be in the case of an overridden `programs.k9s.package`.